Tests/QuranAudioKitTests/QuranAudioRequestBulderTests.swift

```diff
@@ -0,0 +1,164 @@
+//
+//  File.swift
+//  
+//
+//  Created by Mohammad Abdurraafay on 2023-02-11.
+//
+
+import Foundation
+import XCTest
+@testable import QuranAudioKit
+import QuranKit
+
+class GappedAudioRequestBuilderTests: XCTestCase {
+    private var audioRequestBuilder: QuranAudioRequestBuilder!
+    private var reciter: Reciter!
+    
+    override func setUpWithError() throws {
+        reciter = .gappedReciter
+        audioRequestBuilder = GappedAudioRequestBuilder()
+    }
+    
+    override func tearDown() async throws {
+        audioRequestBuilder = nil
+        reciter = nil
+    }
+    
+    func testAudioRequestContainsBismillahWhenThePlaybackIsNotRepeated() throws {
+        let expectation = expectation(description: "waiting for promise to fulfill")
+        let from = try XCTUnwrap(AyahNumber(sura: .secondSura, ayah: Sura.firstSura.firstVerse.ayah))
+        let to = try XCTUnwrap(AyahNumber(sura: .secondSura, ayah: Sura.firstSura.firstVerse.ayah))
+        let totalNumberOfAyah = 1
+        
+        _ = audioRequestBuilder.buildRequest(
+            with: reciter,
+            from: from,
+            to: to,
+            frameRuns: .one,
+            requestRuns: .one
+        ).done { audioRequest in
+            let audioRequest = try XCTUnwrap(audioRequest as? GappedAudioRequest)
+            XCTAssertGreaterThan(audioRequest.ayahs.count, totalNumberOfAyah)
+            expectation.fulfill()
+        }
+        .catch({ error in
+            XCTFail(error.localizedDescription)
+            expectation.fulfill()
+        })
+        wait(for: [expectation], timeout: 40.0)
```

Comment:
Let's use a timeout of 1 second. I don't think the user can wait for 39 seconds for the app to build the AudioRequest. These operations should be sub-second.